chain/chain/src/chain.rs

```diff
@@ -662,6 +655,54 @@ impl Chain {
         })
     }
 
+    /// create a new flat storage state for this shard
```

Comment:
optional: consider expanding a bit what "new" means here in terms the range that will be covered by the new flat state